File: io/io/src/test/resources/compat/issue-756/src/java/LegacyBinaryDecorator.java
package io.bluetape4k.io.serializer.compat.issue756.java;

import io.bluetape4k.io.serializer.BinarySerializer;
import io.bluetape4k.io.serializer.BinarySerializerDecorator;
import java.nio.charset.StandardCharsets;

public final class LegacyBinaryDecorator extends BinarySerializerDecorator {
    public LegacyBinaryDecorator(BinarySerializer serializer) {
        super(serializer);
    }

    @Override
    public byte[] serialize(Object graph) {
        String payload = new String(super.serialize(graph), StandardCharsets.UTF_8);
        return ("decorated:" + payload).getBytes(StandardCharsets.UTF_8);
    }
}

File: io/io/src/test/resources/compat/issue-756/src/kotlin/LegacyBinaryDecorator.kt
package io.bluetape4k.io.serializer.compat.issue756.kotlin

import io.bluetape4k.io.serializer.BinarySerializer
import io.bluetape4k.io.serializer.BinarySerializerDecorator

class LegacyBinaryDecorator(
    serializer: BinarySerializer,
): BinarySerializerDecorator(serializer) {
    override fun serialize(graph: Any?): ByteArray =
        "decorated:${super.serialize(graph).decodeToString()}".encodeToByteArray()
}
